homeassistant/components/igloohome/lock.py

```diff
@@ -0,0 +1,91 @@
+"""Implementation of the lock platform."""
+
+from datetime import timedelta
+
+from aiohttp import ClientError
+from igloohome_api import (
+    BRIDGE_JOB_LOCK,
+    BRIDGE_JOB_UNLOCK,
+    DEVICE_TYPE_LOCK,
+    Api as IgloohomeApi,
+    ApiException,
+    GetDeviceInfoResponse,
+)
+
+from homeassistant.components.lock import LockEntity, LockEntityFeature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import IgloohomeConfigEntry
+from .entity import IgloohomeBaseEntity
+from .utils import get_linked_bridge
+
+# Scan interval set to allow Lock entity update the bridge linked to it.
+SCAN_INTERVAL = timedelta(hours=1)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: IgloohomeConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up lock entities."""
+    async_add_entities(
+        IgloohomeLockEntity(
+            api_device_info=device,
+            api=entry.runtime_data.api,
+            bridge_id=str(
+                get_linked_bridge(device.deviceId, entry.runtime_data.devices)
+            ),
+        )
+        for device in entry.runtime_data.devices
+        if device.type == DEVICE_TYPE_LOCK
+        and get_linked_bridge(device.deviceId, entry.runtime_data.devices) is not None
+    )
+
+
+class IgloohomeLockEntity(IgloohomeBaseEntity, LockEntity):
+    """Implementation of a device that has locking capabilities."""
+
+    # Operating on assumed state because there is no API to query the state.
+    _attr_assumed_state = True
+    _attr_supported_features = LockEntityFeature.OPEN
+
+    def __init__(
+        self, api_device_info: GetDeviceInfoResponse, api: IgloohomeApi, bridge_id: str
+    ) -> None:
+        """Initialize the class."""
+        super().__init__(
+            api_device_info=api_device_info,
+            api=api,
+            unique_key="lock",
+        )
+        self.bridge_id = bridge_id
```

Comment:
```suggestion
        self.bridge_id = bridge_id
        self._attr_name = None
```
is this the main feature of the device? If yes set `_attr_name = None`

Reply:
Hey @zweckj , yes unlocking/locking is the main feature of the device. I'm trying to understand what setting _attr_name does because trying to implement the suggestion provided is causing my snapshot tests to fail with the following error. 

```
tests/components/igloohome/test_lock.py:15 (test_lock)
RegistryEntry(entity_id='lock.front_door', unique_id='lock_OE1X123cbb11', platform='igloohome', previous_unique_id=None, aliases=set(), area_id=None, categories={}, capabilities=None, config_entry_id='01JNWYQXZHP1DHY8SGA0PJFWGC', config_subentry_id=None, created_at=datetime.datetime(2025, 3, 9, 8, 1, 2, 461930, tzinfo=datetime.timezone.utc), device_class=None, device_id='53f2e30993e930a4bf1503d1dc50aa0c', disabled_by=None, entity_category=None, hidden_by=None, icon=None, id='47702189236be5a24d297e1e6c37267c', has_entity_name=True, labels=set(), modified_at=datetime.datetime(2025, 3, 9, 8, 1, 2, 461983, tzinfo=datetime.timezone.utc), name=None, options={}, original_device_class=None, original_icon=None, original_name=None, supported_features=<LockEntityFeature.OPEN: 1>, translation_key=None, unit_of_measurement=None, _cache={}) != SnapshotAssertion(name='snapshot', num_executions=1)

Expected :SnapshotAssertion(name='snapshot', num_executions=1)
Actual   :RegistryEntry(entity_id='lock.front_door', unique_id='lock_OE1X123cbb11', platform='igloohome', previous_unique_id=None, aliases=set(), area_id=None, categories={}, capabilities=None, config_entry_id='01JNWYQXZHP1DHY8SGA0PJFWGC', config_subentry_id=N ...

<Click to see difference>

hass = <HomeAssistant RUNNING>
entity_registry = <homeassistant.helpers.entity_registry.EntityRegistry object at 0xffff6efecc20>
snapshot = SnapshotAssertion(name='snapshot', num_executions=1)
mock_config_entry = <ConfigEntry entry_id=01JNWYQXZHP1DHY8SGA0PJFWGC version=1 domain=igloohome title=Client Credentials state=ConfigEntryState.LOADED unique_id=None>

    async def test_lock(
        hass: HomeAssistant,
        entity_registry: er.EntityRegistry,
        snapshot: SnapshotAssertion,
        mock_config_entry: MockConfigEntry,
    ) -> None:
        """Test lock entity created."""
        with patch("homeassistant.components.igloohome.PLATFORMS", [Platform.LOCK]):
            await setup_integration(hass, mock_config_entry)
    
>       await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)

igloohome/test_lock.py:26:
```

Reply:
Hmm, solved it by updating my snapshot test.